test/api/api_test.go

```diff
@@ -191,10 +191,10 @@ func TestUserTierCache(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	params := url.Values{}
-	params.Add("email", email)
-	params.Add("password", password)
-	r, _, err := at.Post("/login", nil, params)
+	bodyParams := url.Values{}
+	bodyParams.Add("email", email)
+	bodyParams.Add("password", password)
+	r, _, err := at.Post("/login", nil, bodyParams)
```

Comment:
This doesn't actually change the body as suggested by the PR title right? It's still a json object. Only the argumnt to `Post` changed?

Reply:
Not really, no. The PR was created when we changed the handlers to use JSONs and then it was side-tracked by other priorities.

Reply:
Ah I see. In that case wouldn't it make more sense to make the argument for the body a `bodyParams interface{}` ? Then instead of `url.Values` we could use actual types.

e.g. here
```go
type LoginRequest struct {
  Email        string `json:"email"`
  Password     string `json:"password"`
}
```
and then
```go
bodyParams := LoginRequest{
  Email: email,
  Password: password,
}
r, _, err := at.Post("/login", nil, bodyParams)
```

Reply:
We can f/u with it.

You shouldn't need type validation after that because all you do is pass it on to `json.Marshal` anyway.
The additional type safety will come from something like the following:

```go
func (c *Client) Login(email, password string) error {
  return c.post("/login", LoginRequest{
    Email: email,
    Password: password,
  }
}
```

Since `post` is no longer exposed, a user can only call the endpoint through the exposed `Login` method which handles the type nicely.

Reply:
Added to https://github.com/SkynetLabs/skynet-accounts/issues/138